src/app/mytools/my-tool/my-tool.component.ts

```diff
@@ -116,27 +129,53 @@ export class MyToolComponent extends MyEntry implements OnInit {
     this.commonMyEntriesOnInit();
     this.containerService.setTool(null);
     this.containerService.setTools(null);
+    this.workflowService.setWorkflow(null);
+    this.workflowService.setWorkflows(null);
+    this.tool$ = this.toolQuery.tool$;
     this.toolQuery.tool$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((tool) => {
       this.tool = tool;
     });
+    this.workflowQuery.workflow$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((appTool) => {
+      this.appTool = appTool;
+    });
 
     this.getMyEntries();
 
-    this.containerService.tools$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((tools) => {
-      this.tools = tools;
-      this.selectEntry(this.mytoolsService.recomputeWhatEntryToSelect(tools));
-    });
+    combineLatest([this.containerService.tools$, this.workflowService.workflows$])
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe(([tools, workflows]) => {
+        this.tools = tools;
+        this.apptools = workflows;
+        if (this.tools) {
```

Comment:
I think you could simply this block to, unless there is code relying on `this.tools` or `this.workflows` being `null` further on:

```
this.tools = tools || [];
this appTools = workflows || [];
this.allTools = this.tools.concat(this.appTools)
```